Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1

```diff
@@ -594,12 +596,12 @@ function Execute-Main
 {
     Write-Verbose "Entering Execute-Main function" -Verbose
 
-    if(-not (IsInputNullOrEmpty -str $AppPoolName))
+    if($createAppPool -ieq "true")
```

Comment:
is this check required here too?
isn't it duplicate? can we consider this private and assume it's not null - public method always do the check and passes it non null value?